core/Session/SessionAuth.php

```diff
@@ -186,4 +192,24 @@ private function updateSessionExpireTime()
         setcookie(session_name(), session_id(), time() + $sessionCookieLifetime, $sessionParams['path'],
             $sessionParams['domain'], $sessionParams['secure'], $sessionParams['httponly']);
     }
+
+    private function isExpiredSession(SessionFingerprint $sessionFingerprint)
+    {
+        $nonRememberedSessionExpireTime = Config::getInstance()->General['login_session_not_remembered_idle_timeout'];
+        $sessionCookieLifetime = Config::getInstance()->General['login_cookie_expire'];
+
+        $startTime = $sessionFingerprint->getSessionStartTime();
+        if ($startTime === null) {
+            return false;
+        }
+
+        if ($sessionFingerprint->isRemembered()) {
+            $expireTime = $startTime + $sessionCookieLifetime;
+        } else {
+            $expireTime = $startTime + $nonRememberedSessionExpireTime;
+        }
+
+        $isExpired = Date::getNowTimestamp() > $expireTime;
+        return $isExpired;
```

Comment:
btw would we need to destroy the session when it is expired or not important?

Reply:
I don't think we can in PHP, since it requires triggering session GC, and we set different session lifetimes based on whether remember me was checked or not.